src/databricks/labs/lakebridge/contexts/application.py

```diff
@@ -34,9 +35,10 @@ def replace(self, **kwargs):
             self.__dict__[key] = value
         return self
 
-    @cached_property
+    @property
     def workspace_client(self) -> WorkspaceClient:
-        return self._ws
+        w = WorkspaceClient(product="lakebridge", product_version=__version__)
+        return w
```

Comment:
I think this should be left as-is:
```suggestion
        return self._ws
```

Instead, I would prefer it if we added a method to `ApplicationContext` that updates the user-agent, something like:
```python
    def add_user_agent_extra(self, key: str, value: str) -> None:
        ....
```
This should:
1. Create a new `client.Config()` instance using the `.with_user_agent_extra(key, value)` method.
2. Instantiate a new `WorkspaceClient` using the new `Config` and assign it to `self._ws` so that the property just returns the new instance.